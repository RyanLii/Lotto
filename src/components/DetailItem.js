import { View, StyleSheet, Text, FlatList } from 'react-native';
import React, { PureComponent } from 'react';
import moment from 'moment';

export default class DetailItem extends PureComponent {
  _renderItems = (item, isPrimary) => {
    return (
      <View
        style={[
          styles.numberContainer,
          { backgroundColor: isPrimary ? 'red' : 'blue' },
        ]}>
        <Text style={styles.numberText}>{item}</Text>
      </View>
    );
  };

  _renderNumbers = (numbers, isPrimary) => {
    return (
      <FlatList
        horizontal={true}
        data={numbers}
        renderItem={({ item }) => this._renderItems(item, isPrimary)}
        keyExtractor={(item, index) => {
          return index.toString();
        }}
      />
    );
  };

  render() {
    const {
      DrawDate,
      DrawDisplayName,
      DrawLogoUrl,
      PrimaryNumbers,
      SecondaryNumbers,
    } = this.props;
    const date = moment(DrawDate).format('DD');
    const month = moment(DrawDate).format('MMMM');
    const year = moment(DrawDate).format('yyyy');
    return (
      <View style={styles.container}>
        <View style={styles.columnContainer}>
          <View style={styles.dateContainer}>
            <Text style={styles.dayText}>{date}</Text>
            <View style={styles.columnContainer}>
              <Text>{month}</Text>
              <Text>{year}</Text>
            </View>
          </View>
          <Text>{DrawDisplayName}</Text>
        </View>
        <View style={([styles.columnContainer], { marginLeft: 20 })}>
          {this._renderNumbers(PrimaryNumbers, true)}
          {this._renderNumbers(SecondaryNumbers, false)}
        </View>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 10,
    marginVertical: 8,
    marginHorizontal: 16,
    alignItems: 'center',
  },
  dateContainer: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  columnContainer: {
    flexDirection: 'column',
  },
  dayText: {
    fontSize: 30,
    padding: 5,
    fontWeight: 'bold',
  },
  numberContainer: {
    height: 30,
    width: 30,
    borderRadius: 15,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'red',
  },
  numberText: {
    color: 'white',
  },
});
