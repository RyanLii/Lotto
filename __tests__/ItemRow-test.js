import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import HomepageItem from '../src/components/HomepageItem';
import DetailItem from '../src/components/DetailItem';

test('renders correctly', () => {
  const tree = renderer.create(<HomepageItem />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders DetailItem correctly', () => {
  const tree = renderer.create(<DetailItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
