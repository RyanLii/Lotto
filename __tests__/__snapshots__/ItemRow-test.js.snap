// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DetailItem correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginHorizontal": 16,
      "marginVertical": 8,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "padding": 5,
          }
        }
      >
        06
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text>
          September
        </Text>
        <Text>
          2020
        </Text>
      </View>
    </View>
    <Text />
  </View>
  <View
    style={
      Object {
        "marginLeft": 20,
      }
    }
  >
    <RCTScrollView
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
    <RCTScrollView
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "beige",
      "flexDirection": "row",
      "marginHorizontal": 16,
      "marginVertical": 8,
      "padding": 10,
    }
  }
>
  <Image
    resizeMode="contain"
    style={
      Object {
        "borderRadius": 25,
        "height": 50,
        "margin": 10,
        "width": 50,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</View>
`;
